components/tabs/src/storage.rs

```diff
@@ -20,13 +20,16 @@ use sql_support::open_database::{self, open_database_with_flags};
 use sql_support::ConnExt;
 use std::cell::RefCell;
 use std::collections::HashMap;
+use std::mem;
 use std::path::{Path, PathBuf};
 use sync15::{RemoteClient, ServerTimestamp};
 pub type TabsDeviceType = crate::DeviceType;
 pub type RemoteTabRecord = RemoteTab;
 
 pub(crate) const TABS_CLIENT_TTL: u32 = 15_552_000; // 180 days, same as CLIENTS_TTL
 const FAR_FUTURE: i64 = 4_102_405_200_000; // 2100/01/01
+const MAX_PAYLOAD_SIZE: usize = 256 * 1024; // Same as Desktop, see service.js or sync15/request.rs
```

Comment:
I believe desktop is stale in this regard, and that tabs have the same max-payload as anything else, which inspecting info/configuration tells me is `max_record_payload_bytes: 2097152`, so much bigger than this. It's actually super unfortunate that the sync engine doesn't have this available to it so it can't easily be passed as a param. I'm not sure if we need to go all the way to 2MB though, but it does seem unfortunately to be this small if we don't need to and users are actually not seeing all the tabs they want. I'll ask on #sync

Reply:
Per the conversation, we don't seem to have an agreed amount vs just going to the max -- I decided to put it at 1MB since it's (1) > 256k and (2) < 2MB, but obviously it's super easy to change.